test/server/util/api.test.js

```diff
@@ -0,0 +1,166 @@
+const { asyncMiddleware } = require('../../../src/server/util/api');
+
+describe('util/api', () => {
+    describe('asyncMiddleware', () => {
+        test('two param middleware', async () => {
+            const mockReq = {};
+            const mockRes = {};
+
+            expect.assertions(2);
+
+            const testMiddle = async (req, res) => {
+                expect(req).toEqual(mockReq);
+                expect(res).toEqual(mockRes);
+            };
+
+            const wrapped = asyncMiddleware(testMiddle);
+
+            await wrapped(mockReq, mockRes);
+        });
+
+        test('two param middleware throws', async () => {
+            const mockReq = {};
+            const mockRes = {};
+            const mockError = new Error('mock');
+
+            expect.assertions(4);
+
+            const testMiddle = async (req, res) => {
+                expect(req).toEqual(mockReq);
+                expect(res).toEqual(mockRes);
+                throw mockError;
+            };
+
+            const mockNext = jest.fn(error => {
+                expect(error).toBe(mockError);
+            });
+
+            const wrapped = asyncMiddleware(testMiddle);
+
+            expect(wrapped.length).toBe(3);
+
+            // express will still call it with 3 params cuz asyncMiddleware gives it 3 params
+            await wrapped(mockReq, mockRes, mockNext);
+        });
+
+        test('three param middleware', async () => {
+            const mockReq = {};
+            const mockRes = {};
+            const mockNext = jest.fn();
+
+            expect.assertions(3);
+
+            const testMiddle = async (req, res, next) => {
+                expect(req).toEqual(mockReq);
+                expect(res).toEqual(mockRes);
+                expect(next).toEqual(mockNext);
+            };
+
+            const wrapped = asyncMiddleware(testMiddle);
+
+            await wrapped(mockReq, mockRes, mockNext);
+        });
+
+        test('three param middleware throws', async () => {
+            const mockReq = {};
+            const mockRes = {};
+            const mockError = new Error('mock');
+
+            expect.assertions(5);
+
+            const testMiddle = async (req, res, next) => {
+                expect(req).toEqual(mockReq);
+                expect(res).toEqual(mockRes);
+                expect(next).toEqual(mockNext);
+                throw mockError;
+            };
+
+            const mockNext = jest.fn(error => {
```

Comment:
Would this be helpful to declare before it's used on line 74? Or is that referencing a different `mockNext`?

Reply:
This is an eslint rule somewhere -- we should enable it and enforce commenting weird situations that require use-before-def.

But yeah - I'll move this up when I update lint for it, it functions the same since `testMiddle` isn't executed until ater.